File: frontend/src/model/Game.ts
import Rating from './Rating';
import Console from './Console';

export default interface Game {
    id: string,
    name: string,
    developer: string,
    releaseDate: Date,
    releaseDateView: string,
    ratings: Rating[],
    picturePath: string,
    consoles: Console[],
}

File: backend/src/main/kotlin/lise/uebungsprojekt/model/Game.kt
package lise.uebungsprojekt.model

import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.databind.ser.std.ToStringSerializer
import org.bson.types.ObjectId
import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import java.util.Date

@Document
data class Game(@Id @JsonSerialize(using = ToStringSerializer::class) override val id: ObjectId,
                override val name: String, override val releaseDate: Date, override var releaseDateView: String?,
                override val developer: String, override var ratings: List<Rating>, override var averageRating: Double?,
                override val picturePath: String, override val consoles: List<Console>): GameBase()